pkg/controller/drain/drain_controller.go

```diff
@@ -381,9 +456,34 @@ func (ctrl *Controller) drainNode(node *corev1.Node, drainer *drain.Helper) erro
 			ctrl.enqueueAfter(node, ctrl.cfg.DrainRequeueDelay)
 		}
 
+		nErr := upgrademonitor.GenerateAndApplyMachineConfigNodes(
+			&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateExecuted, Reason: string(v1alpha1.MachineConfigNodeUpdateDrained), Message: fmt.Sprintf("Node Drain has failed")},
+			&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateDrained, Reason: fmt.Sprintf("%s%s", string(v1alpha1.MachineConfigNodeUpdateExecuted), string(v1alpha1.MachineConfigNodeUpdateDrained)), Message: fmt.Sprintf("Node Drain has failed. Error is: %s The drain will not be complete until desired drainer %s matches current drainer %s", err.Error(), node.Annotations[constants.DesiredDrainerAnnotationKey], node.Annotations[constants.LastAppliedDrainerAnnotationKey])},
+			metav1.ConditionUnknown,
+			metav1.ConditionUnknown,
+			node,
+			ctrl.client,
+			ctrl.featureGatesAccessor,
+		)
+		if nErr != nil {
+			klog.Errorf("Error making MCN: %w", nErr)
+		}
+
 		// Return early without deleting the ongoing drain.
 		return err
 	}
+	err = upgrademonitor.GenerateAndApplyMachineConfigNodes(
+		&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateExecuted, Reason: string(v1alpha1.MachineConfigNodeUpdateDrained), Message: fmt.Sprintf("Draining Node as part of In progress upgrade phase")},
```

Comment:
I think we should use `in progress update phase` instead of `in progress upgrade phase`. Upgrade is a subset of when we perform drain. Drain also occurs when we do other basic operation like writing files/ services.

Reply:
Ok, I can change all references